### PR TITLE
Double trans v2

### PR DESCRIPTION
**Pull request by JherezTaylor**
@NicholasDanks @rjollet- managed to build it using matrices. Check it out and let me know what you think

### CODE REVIEW

File: double_trans_cipher.rb [hunk 1]
```diff
@@ -1,15 +1,59 @@
+## DoubleTranspositionCipher
 module DoubleTranspositionCipher
+  def self.row_swap(origin, dest, random_key)
+    origin.each_with_index do |_row, i|
+      dest[random_key[i]] = origin[i]
+    end
+    dest
+  end
+
+  def self.undo_swap(origin, dest, random_key)
```
Comment (rjollet) [inline double_trans_cipher.rb:10]: why do you put dest as a parameter ?

File: double_trans_cipher.rb [hunk 2]
```diff
@@ -1,15 +1,59 @@
...
+
+    origin_matrix, random_key, col = init_matrix(document, key)
+
+    row_sorted_matrix = Array.new(col) { Array.new(col, 0) }
+    col_sorted_matrix = Array.new(col) { Array.new(col, 0) }
+
+    row_sorted_matrix = row_swap(origin_matrix, row_sorted_matrix, random_key)
+    row_sorted_matrix = row_sorted_matrix.transpose
+
+    col_sorted_matrix = row_swap(row_sorted_matrix, col_sorted_matrix, random_key)
```
Comment (rjollet) [inline double_trans_cipher.rb]: you can do like:

```
col_sorted_matrix = row_swap(row_sorted_matrix.transpose, col_sorted_matrix, random_key)
```

and then delete the line 40

File: double_trans_cipher.rb (hunk 1 above)
Comment (JherezTaylor) [inline double_trans_cipher.rb:10]: It's an empty matrix of the same size that I'm passing in, Ruby doesn't allow you to deep copy arrays. So I couldn't do a modification of the original array without loosing values that I needed

Comment (rjollet) [inline double_trans_cipher.rb:10]: oh ok I got it

File: double_trans_cipher.rb (hunk 2 above)
Comment (JherezTaylor) [inline double_trans_cipher.rb]: Yea, that's how I had it originally but rubocop was complaining about the length of the line so I just changed it in the end

Comment (JherezTaylor) [inline double_trans_cipher.rb]: But he says we can ignore line length so I will go back in and tidy it up

**Comment by rjollet** [comment]
Ok it's all good for me

**Pull request merged**